src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -263,9 +263,16 @@ private void ConsumeFromRemainingBuffer(int bytesToConsume)
                         cookiesFromContainer = null;
                     }
 
+                    HttpHeaderParser parser = header.Key.Parser;
```

Comment:
Please add a comment about *why* this is necessary, and specifically mention the ProductHeaderInfo or whatever that class is called.